include/phasar/Utils/MemoryResource.h

```diff
@@ -2,7 +2,7 @@
 /// a workaround here
 
 #ifndef HAS_MEMORY_RESOURCE
-#if !defined(__has_include) || __has_include(<memory_resource>)
+#if !defined(__has_include) || __has_include(<memory_resource>) && !defined(__APPLE__)
```

Comment:
We should probably add a comment to describe why we exclude pmr on apple.

Btw, are the feature-test macros `__cpp_lib_memory_resource` and `__cpp_lib_polymorphic_allocator` defined on apple?